catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -109,22 +117,64 @@ public PolicyResponse processPostQuery(Result input, Map<String, Serializable> p
       return new PolicyResponseImpl();
     }
     Metacard metacard = input.getMetacard();
-    HashMap<String, Set<String>> securityMap = new HashMap<>();
 
+    Map<String, Set<String>> securityMap = new HashMap<>();
     if (isMarkedForFilter(filterErrors, metacard, Validation.VALIDATION_ERRORS)
         || isMarkedForFilter(filterWarnings, metacard, Validation.VALIDATION_WARNINGS)) {
-      for (Map.Entry<String, List<String>> attributeMapping : attributeMap.entrySet()) {
-        securityMap.put(attributeMapping.getKey(), new HashSet<>(attributeMapping.getValue()));
-      }
+      securityMap = getSecurityMapConfiguration();
     }
 
     return new PolicyResponseImpl(new HashMap<>(), securityMap);
   }
 
-  private boolean isMarkedForFilter(boolean filterFlag, Metacard metacard, String attribute) {
-    return filterFlag
-        && metacard.getAttribute(attribute) != null
-        && metacard.getAttribute(attribute).getValues() != null;
+  /**
+   * Modifies the incoming {@link QueryRequest}s filter depending upon the configuration for this
+   * plugin. If a {@link Subject} contains the appropriate configured roles, it will always receive
+   * {@link Metacard}s with errors and/or warnings, otherwise the {@link Subject} will receive
+   * results based on the {@code filterErrors} and {@code filterWarnings} flags.
+   *
+   * @param source {@link Source} being queried
+   * @param input {@link QueryRequest} being sent to the {@link Source}
+   * @return new {@link QueryRequest} if the filter was modified, otherwise the original {@link
+   *     QueryRequest}
+   * @throws StopProcessingException if the {@link Subject} is not available
+   */
+  @Override
+  public QueryRequest process(Source source, QueryRequest input) throws StopProcessingException {
+    if (!isLocalSource(source)) {
```

Comment:
If the query does not already contain a filter for warning/error attributes, each source will be responsible for determining whether or not it wants to send Metacards with errors/warnings back depending on that systems configuration for this plugin. This is how it was before, and I think it was because not each endpoint supports these filters.